packages/react-components/src/layout/src/Flex.tsx

```diff
@@ -1,102 +1,70 @@
      */
-    alignContent: oneOf([
-        "start",
-        "end",
-        "center",
-        "space-between",
-        "space-around",
-        "space-evenly",
-        "stretch",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    alignContent?: "start" | "end" | "center" | "space-between" | "space-around" | "space-evenly" | "stretch" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center";
     /**
      * The alignment of children within their container. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/align-items).
      */
-    alignItems: oneOf([
-        "start",
-        "end",
-        "center",
-        "stretch",
-        "self-start",
-        "self-end",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    alignItems?: "start" | "end" | "center" | "stretch" | "self-start" | "self-end" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center";
     /**
      * The distribution of space around items along the main axis. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content).
      */
-    justifyContent: oneOf([
-        "start",
-        "end",
-        "center",
-        "left",
-        "right",
-        "space-between",
-        "space-around",
-        "space-evenly",
-        "stretch",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    justifyContent?: "start" | "end" | "center" | "left" | "right" | "space-between" | "space-around" | "space-evenly" | "stretch" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center"
+    /**
+     * TODO: document this property
+     */
+    noGap?: boolean;
     /**
      * The space between both rows and columns. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/gap).
      */
-    gap: oneOfType([oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]), string]),
+    gap?: (1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13) | string;
     /**
      * Whether flex items are forced onto one line or can wrap onto multiple lines. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-wrap).
      */
-    wrap: oneOf(["nowrap", "wrap", "wrap-reverse"]),
+    wrap?: "nowrap" | "wrap" | "wrap-reverse";
     /**
      * Whether the elements take up all the space of their container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * Whether to wrap children in a `div` element.
      */
-    wrapChildren: bool,
+    wrapChildren?: boolean;
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
-     * @ignore
```

Comment:
I would document children properties when they are required, what do you think?

Reply:
Of course!